compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1584,6 +1585,7 @@ public void visit(BLangTupleVariable varNode, AnalyzerData data) {
             varNode.setBType(symResolver.resolveTypeNode(varNode.typeNode, currentEnv));
         }
 
+        analyzeNode(varNode.typeNode, data);
```

Comment:
Do we need this for error binding patterns also?

Reply:
The error is not logged for

```ballerina
public function main() {
    error<map<[@UndefinedAnnotation int]>> error () = error("err!");
}
```

Reply:
Didn't log error for following scenario as well
```ballerina
public function main() {
    error<record {|@UndefinedAnnotation int x = 10;|}> error () = error("err!");
}
```

Reply:
Fixed.